test/distributed/test_c10d_nccl.py

```diff
@@ -3582,6 +3585,42 @@ def test_short(self):
         before_test = now - timedelta(minutes=1)
         self.assertTrue(before_test < event_created_time < now)
 
+    @requires_nccl()
+    @skip_but_pass_in_sandcastle_if(torch.cuda.device_count() < 2, "NCCL test requires 2+ GPUs")
+    def test_dump_pipe(self):
+        def open_file_with_timeout(file_path, mode, timeout=1.0):
+            start_time = time.time()
+            while time.time() - start_time < timeout:
+                if os.path.exists(file_path):
```

Comment:
do you worry about race between the file existing and having some bytes and the file being closed?  (will this test be unreliable and get partial file sometimes)

Reply:
is this solved by 'unpickle'?  hopefully pickle won't unpickle an incomplete file?

Reply:
Yeah, it should be avoided by unpickle continuing to read the pickle until it hits the end of the pickle, which is why I directly connected it to the file.